[build-system]
requires = ["setuptools", "setuptools_scm"]
build-backend = "setuptools.build_meta"


[project]
name = "natural-language-geocoding"
authors = [{ name = "Jason Gilman", email = "jason@element84.com" }]
maintainers = [{ name = "Jason Gilman", email = "jason@element84.com" }]
description = "Natural Language Geocoding implements an AI technique to convert natural language descriptions of spatial areas into polygons."
readme = "README.md"
requires-python = ">=3.10"
license = { file = "LICENSE" }
dependencies = [
  # TODO once it's in pypi enable this
  # "e84-geoai-common>=0.0.1"

  "requests>=2.32.3"
]
dynamic = ["version"]

[tool.setuptools_scm]

[project.urls]
Github = "https://github.com/Element84/natural-language-geocoding"

[project.scripts]
natural-language-geocoding = "natural_language_geocoding.main:main"

[project.optional-dependencies]
demo = [
  # Must use this version of Gradio until this issue is resolved in Gradio Folium
  # https://huggingface.co/spaces/freddyaboulton/gradio_folium/discussions/3
  "gradio==4.40.0",
  "gradio-folium>=0.0.6",

  # Using this version of fastapi due to a pydantic issue.
  # See https://discuss.huggingface.co/t/new-gradio-space-connection-errored-out/105509
  # and https://github.com/oobabooga/text-generation-webui/issues/6367
  # Latest version with the problem is 0.115.0. Try after they release a new version.
  "fastapi==0.112.2"
]
dev = [
  "pytest>=8.3.3",
  "ipykernel>=6.29.5",
  "black>=24.8.0",
  "pyright>=1.1.381",
  "build>=1.2.2",
  "pre-commit>=3.8.0",
  "pre-commit-hooks>=4.6.0"
]
